src/Cooked/MockChain/BlockChain.hs

```diff
@@ -381,6 +382,33 @@ txSkelInputDataAsHashes skel = do
   (Map.elems -> inputTxOuts) <- txSkelInputUtxos skel
   catMaybes <$> mapM outputToDatumHashM inputTxOuts
 
+-- | This creates a payment from an existing TxOut. This is not trivial because
+-- we need to reconstruct a bunch of information using data withing the blockchain
+txOutRefToTxSkelOut :: (MonadBlockChainBalancing m) => Api.TxOutRef -> m TxSkelOut
+txOutRefToTxSkelOut oRef = do
+  Just txOut@(Api.TxOut (Api.Address cred _) _ dat refS) <- txOutByRef oRef
+  target <- case cred of
+    Api.PubKeyCredential pkh -> return $ Left pkh
+    Api.ScriptCredential (Api.ScriptHash sh) -> do
+      Just val <- validatorFromHash (Script.ValidatorHash sh)
+      return $ Right val
+  datum <- case dat of
+    Api.NoOutputDatum -> return TxSkelOutNoDatum
+    Api.OutputDatumHash hash -> do
+      Just (Api.Datum dat') <- datumFromHash hash
+      return $ TxSkelOutDatum dat' -- TODO: investigate between this and TxSkelOutDatum
```

Comment:
There seems to be a typo in this TODO remark because `TxSkelOutDatum` is what is already used here.